### ISSUE TITLE
BUG: cannot use `faster_rcnn` in ObjectDetector

### ISSUE THREAD

**Issue by IncubatorShokuhou** [PyTorchLightning/lightning-flash#1265]
## 🐛 Bug
```
Traceback (most recent call last):
  File "/mnt/data/buduanban_detection/test.py", line 25, in <module>
    trainer.finetune(model, datamodule=datamodule, strategy="freeze")
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/lightning_flash-0.8.0.dev0-py3.9.egg/flash/core/trainer.py", line 161, in finetune
    return super().fit(model, train_dataloader, val_dataloaders, datamodule)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 740, in fit
    self._call_and_handle_interrupt(
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 685, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 777, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1199, in _run
    self._dispatch()
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1279, in _dispatch
    self.training_type_plugin.start_training(self)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 202, in start_training
    self._results = trainer.run_stage()
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1289, in run_stage
    return self._run_train()
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1311, in _run_train
    self._run_sanity_check(self.lightning_module)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1375, in _run_sanity_check
    self._evaluation_loop.run()
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/loops/base.py", line 145, in run
    self.advance(*args, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/loops/dataloader/evaluation_loop.py", line 110, in advance
    dl_outputs = self.epoch_loop.run(dataloader, dataloader_idx, dl_max_batches, self.num_dataloaders)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/loops/base.py", line 145, in run
    self.advance(*args, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py", line 122, in advance
    output = self._evaluation_step(batch, batch_idx, dataloader_idx)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py", line 217, in _evaluation_step
    output = self.trainer.accelerator.validation_step(step_kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/accelerators/accelerator.py", line 239, in validation_step
    return self.training_type_plugin.validation_step(*step_kwargs.values())
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 219, in validation_step
    return self.model.validation_step(*args, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/lightning_flash-0.8.0.dev0-py3.9.egg/flash/core/adapter.py", line 113, in validation_step
    return self.adapter.validation_step(batch, batch_idx)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/lightning_flash-0.8.0.dev0-py3.9.egg/flash/core/integrations/icevision/adapter.py", line 233, in validation_step
    return self.icevision_adapter.validation_step(batch[DataKeys.INPUT], batch_idx)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/icevision/models/torchvision/lightning/model_adapter.py", line 41, in validation_step
    train_preds = self(xb, yb)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/icevision/models/torchvision/lightning/model_adapter.py", line 26, in forward
    return self.model(*args, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torchvision/models/detection/generalized_rcnn.py", line 96, in forward
    proposals, proposal_losses = self.rpn(images, features, targets)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torchvision/models/detection/rpn.py", line 344, in forward
    anchors = self.anchor_generator(images, features)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torchvision/models/detection/anchor_utils.py", line 125, in forward
    anchors_over_all_feature_maps = self.grid_anchors(grid_sizes, strides)
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/torchvision/models/detection/anchor_utils.py", line 86, in grid_anchors
    raise ValueError("Anchors should be Tuple[Tuple[int]] because each feature "
ValueError: Anchors should be Tuple[Tuple[int]] because each feature map could potentially have different sizes and aspect ratios. There needs to be a match between the number of feature maps passed and the number of sizes / aspect ratios specified.
```
### To Reproduce
Change the head and backbone of [example in the doc](https://lightning-flash.readthedocs.io/en/latest/reference/object_detection.html) to  `head="faster_rcnn", backbone="mobilenet" `
#### Code sample
```python
import flash
from flash.core.data.utils import download_data
from flash.image import ObjectDetectionData, ObjectDetector

# 1. Create the DataModule
# Dataset Credit: https://www.kaggle.com/ultralytics/coco128
# download_data("https://github.com/zhiqwang/yolov5-rt-stack/releases/download/v0.3.0/coco128.zip", "data/")

datamodule = ObjectDetectionData.from_coco(
    train_folder="data/coco128/images/train2017/",
    train_ann_file="data/coco128/annotations/instances_train2017.json",
    val_split=0.1,
    transform_kwargs={"image_size": 512},
    batch_size=4,
)

# 2. Build the task
model = ObjectDetector(head="faster_rcnn",   #  <----  use faster_rcnn
                       backbone="mobilenet",     #  <----  use mobilenet
                       num_classes=datamodule.num_classes, 
                       image_size=512)

# 3. Create the trainer and finetune the model
trainer = flash.Trainer(max_epochs=1)
trainer.finetune(model, datamodule=datamodule, strategy="freeze")

# 4. Detect objects in a few images!
datamodule = ObjectDetectionData.from_files(
    predict_files=[
        "data/coco128/images/train2017/000000000625.jpg",
        "data/coco128/images/train2017/000000000626.jpg",
        "data/coco128/images/train2017/000000000629.jpg",
    ],
    transform_kwargs={"image_size": 512},
    batch_size=4,
)
predictions = trainer.predict(model, datamodule=datamodule)
print(predictions)

# 5. Save the model!
trainer.save_checkpoint("object_detection_model.pt")
```

### Environment

 - OS (e.g., Linux):
 - Python version:
 - PyTorch/Lightning/Flash Version (e.g., 1.10/1.5/0.7):
 - GPU models and configuration:
 - Any other relevant information:

**Comment by IncubatorShokuhou** [comment]
BTW, all backbones in `faster_rcnn` has the same bug.

**Comment by ethanwharris** [comment]
Hey @IncubatorShokuhou Thanks for reporting this! I can reproduce the error, looking in to it 😃

**Comment by ethanwharris** [comment]
@IncubatorShokuhou Just to give you a bit of an update, the issue only relates to non-fpn backbones. So any backbone with `fpn` in the name should work. E.g. `resnet18_fpn` is working fine for me. Working on a fix for the other backbones 😃

**Comment by ethanwharris** [comment]
Turns out that the non-fpn backbones are not intended to work with the object detection / instance segmentation / keypoint detection models from icevision (see discussion: https://github.com/airctic/icevision/issues/1066). So the fix in #1267 is to no longer show those backbones to the user :smiley:
